torch/_prims/nvfuser_prims.py

```diff
@@ -0,0 +1,301 @@
+
+
+for fname in _nvfuser_unary_ops:
+    exec(
+        f"""
+# Ensure that the nvfuser implementation exists
+_assert_nvfuser_op_exists("{fname}")
+
+def _{fname}_nvfuser(fd, a):
+    return fd.Ops.{fname}(a)  # type: ignore[attr-defined]
+
+_nvfuser_impls["{fname}"] = _{fname}_nvfuser
+"""
+    )
+
+_nvfuser_binary_ops = {
+    "add",
+    "atan2",
+    "bitwise_and",
+    "bitwise_or",
+    "bitwise_xor",
+    "div",
+    "eq",
+    "fmod",
+    "ge",
+    "gt",
+    "le",
+    "lt",
+    "mul",
+    "ne",
+    "pow",
+    "remainder",
+    "sub",
+}
+
+for fname in _nvfuser_binary_ops:
+    exec(
+        f"""
+# Ensure that the nvfuser implementation exists
+_assert_nvfuser_op_exists("{fname}")
+
+def _{fname}_nvfuser(fd, a, b):
+    return fd.Ops.{fname}(a, b)  # type: ignore[attr-defined]
+
+_nvfuser_impls["{fname}"] = _{fname}_nvfuser
+"""
+    )
+
+_nvfuser_ternary_ops = {
+    "where",
+}
+
+for fname in _nvfuser_ternary_ops:
+    exec(
+        f"""
+# Ensure that the nvfuser implementation exists
+_assert_nvfuser_op_exists("{fname}")
+
+def _{fname}_nvfuser(fd, a, b, c):
+    return fd.Ops.{fname}(a, b, c)  # type: ignore[attr-defined]
+
+_nvfuser_impls["{fname}"] = _{fname}_nvfuser
+"""
+    )
+
+
+def _broadcast_in_dim_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    shape: ShapeType,
+    broadcast_dimensions: ShapeType,
+):
+    return fd.Ops.broadcast_in_dim(a, shape, broadcast_dimensions)  # type: ignore[attr-defined]
+
+
+def _convert_element_type_nvfuser(fd: Any, a: TensorLikeType, dtype: torch.dtype):
+    nvfuser_dtype = getnvFuserDtype(dtype)
+    return fd.Ops.cast(nvfuser_dtype, a)  # type: ignore[attr-defined]
+
+
+def _sum_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    dims: DimsSequenceType,
+):
+    keep_dims = False
+    output_dtype = torch._C._nvfuser.DataType.Null
+    return fd.Ops.sum(a, dims, keep_dims, output_dtype)
+
+
+def _var_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    dims: DimsSequenceType,
+    *,
+    correction: int,
+):
+    keep_dims = False
+    return fd.Ops.var(a, dims, correction, keep_dims)
+
+
+def _amax_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    dims: DimsSequenceType,
+):
+    keep_dims = False
+    return fd.Ops.max(a, dims, keep_dims)
+
+
+def _amin_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    dims: DimsSequenceType,
+):
+    keep_dims = False
+    return fd.Ops.min(a, dims, keep_dims)
+
+
+_nvfuser_impls["broadcast_in_dim"] = _broadcast_in_dim_nvfuser
+_nvfuser_impls["convert_element_type"] = _convert_element_type_nvfuser
+_nvfuser_impls["sum"] = _sum_nvfuser
+_nvfuser_impls["var"] = _var_nvfuser
+_nvfuser_impls["amax"] = _amax_nvfuser
+_nvfuser_impls["amin"] = _amin_nvfuser
+
+
+def register_nvprims():
+    """Registers all nvFuser primitives in the torch.ops.nvprims module."""
+    for name in nvprim_names:
+        main_prim = getattr(torch.ops.prims, name)
+
+        nvprim.define(main_prim.schema)
+        nvprim_impl.impl(name, main_prim.prim_impl)
```

Comment:
Ha, I see why nvprim is currently implemented as a subset of prim now.

I guess this is only temporary in this PR since we fallback to prim for aten/meta implementations. Please discard my previous comment.